src/odemis/driver/tescan.py

```diff
@@ -884,38 +1077,62 @@ def terminate(self):
     """
     This is an extension of model.Detector class. It performs the main functionality
     of the SEM. It sets up a Dataflow and notifies it every time that an SEM image
     is captured.
     """
-    def __init__(self, name, role, parent, channel, detector, **kwargs):
+    def __init__(self, name, role, parent, channel, detector, scanner, **kwargs):
         """
         channel (0<= int): input channel from which to read
         detector (0<= int): detector index
         """
         # It will set up ._shape and .parent
         model.Detector.__init__(self, name, role, parent=parent, **kwargs)
+        self._scanner = scanner
+        self._device_handler = self._scanner._device_handler
+
         self._channel = channel
-        self._detector = detector
-        self.parent._device.DtSelect(self._channel, self._detector)
-        self.parent._device.DtEnable(self._channel, 1, 16)  # 16 bits
-        # adjust brightness and contrast
-        self.parent._device.DtAutoSignal(self._channel)
+        self._detector = self.get_detector_idx(detector)
+        self._device_handler.DtSelect(self._channel, self._detector)
 
-        # The shape is just one point, the depth
+        # 8 or 16 bits image
+        self.bpp = model.IntEnumerated(DEFAULT_BITDEPTH, {8, 16}, unit="bit", setter=self._setBpp)
         self._shape = (2 ** 16,)  # only one point
+
+        # will take care of executing autocontrast asynchronously
+        self._executor = CancellableThreadPoolExecutor(max_workers=1)  # one task at a time
+
         self.data = SEMDataFlow(self, parent)
 
         # Special event to request software unblocking on the scan
         self.softwareTrigger = model.Event()
 
-        # TODO: provide a method applyAutoContrast(), as in Phenom, to run the
-        # auto signal function. + a way to do so even if the detector is not
-        # used (because it's used via a CompositedScanner)?
+    def get_detector_idx(self, detector: Union[int, str]) -> int:
+        """Get the index of the detector by int (do nothing) or by name (match with Tescan's available detectors).
+
+        :param detector: the detector index or the name of the detector
+        :returns: the detector's index on the Tescan hardware
+        """
+        detector_idx = None
+        if isinstance(detector, int):
+            detector_idx = detector
+        elif isinstance(detector, str):
+            available_detectors = self._device_handler.DtEnumDetectors()
+            # First find the enumeration index for the desired detector (by name)
+            enum_idx_for_name = re.findall(rf"det.([0-9])+.name={detector.lower()}", available_detectors.lower())
+            if enum_idx_for_name:
+                enum_idx_for_name = enum_idx_for_name[0]
+                # Now find the corresponding detector idx (which is not the same as the enumeration index!).
+                det_idx_for_name = re.findall(rf"det.{enum_idx_for_name}+.detector=([0-9]+)", available_detectors)
```

Comment:
I don't think the `+` in `{enum_idx_for_name}+` makes sense. It'd be replaced by something like 1+, which could match 11 too. => drop the `+`